Classes/Audio/VorbisSpectrogramGenerator.cs

```diff
@@ -20,12 +21,44 @@ public class VorbisSpectrogramGenerator {
 
     private CancellationTokenSource tokenSource;
     private string filePath;
+    // Settings
+    private SpectrogramType type;
+    private int maxFreq;
+    private string colormap;
+    private bool drawFlipped;
     private bool isDrawing;
-    public VorbisSpectrogramGenerator(string filePath) {
+    // Since the BMP generated for spectrogram doesn't depend on the component height/width, we can cache it to save on calculations.
+    private bool cache;
+    private ImageSource cachedSpectrogram;
+    private string cachedBmpSpectrogramPath;
+
+    public VorbisSpectrogramGenerator(string filePath, bool cache, SpectrogramType? type, int? maxFreq, String colormap, bool drawFlipped) {
         RecreateTokens();
         this.filePath = filePath;
+        InitSettings(cache, type, maxFreq, colormap, drawFlipped);
         this.isDrawing = false;
     }
+
+    public void InitSettings(bool cache, SpectrogramType? type, int? maxFreq, String colormap, bool drawFlipped) {
+        this.cache = cache;
```

Comment:
This should probably cancel previous drawing process - now it's possible to change the settings multiple times in a row while the previous image is still being generated and get the old image saved under new filename.